gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -90,42 +90,37 @@ case class TransformPreOverrides(isAdaptiveContext: Boolean)
    */
   private def genHashAggregateExec(plan: HashAggregateExec): SparkPlan = {
     val newChild = replaceWithTransformerPlan(plan.child)
+    def transformHashAggregate(): GlutenPlan = {
+      BackendsApiManager.getSparkPlanExecApiInstance
+        .genHashAggregateExecTransformer(
+          plan.requiredChildDistributionExpressions,
+          plan.groupingExpressions,
+          plan.aggregateExpressions,
+          plan.aggregateAttributes,
+          plan.initialInputBufferOffset,
+          plan.resultExpressions,
+          newChild
+        )
+    }
+
     // If child's output is empty, fallback or offload both the child and aggregation.
     if (plan.child.output.isEmpty && BackendsApiManager.getSettings.fallbackAggregateWithChild()) {
       newChild match {
         case _: TransformSupport =>
           // If the child is transformable, transform aggregation as well.
           logDebug(s"Columnar Processing for ${plan.getClass} is currently supported.")
-          BackendsApiManager.getSparkPlanExecApiInstance
-            .genHashAggregateExecTransformer(
-              plan.requiredChildDistributionExpressions,
-              plan.groupingExpressions,
-              plan.aggregateExpressions,
-              plan.aggregateAttributes,
-              plan.initialInputBufferOffset,
-              plan.resultExpressions,
-              newChild
-            )
+          transformHashAggregate()
+        case i: InMemoryTableScanExec if i.supportsColumnar =>
```

Comment:
Thanks so much for fix!
Shouldn't we add condition `BackendsApiManager.isVeloxBackend && GlutenConfig.getConf.columnarTableCacheEnabled` to make sure this is a scan for table cached by Gluten (i.e., child offloaded case)?

Reply:
yes, we have these checks, `InMemoryTableScanExec.supportsColumnar` will call `CachedBatchSerializer.supportsColumnarOutput`. So if supportsColumnar is true, that means we are in velox backend and enabled columnar table cache.

Reply:
Can it be vanilla reader for cached table when `supportsColumnar` is true?
I note this part of code makes agg plan & its child fall back or offload together. Is it necessary to add more check?
In addition, I note there is some check [here](
https://github.com/oap-project/gluten/blob/main/gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala#L707) for `InMemoryTableScanExec`. Just wondering why they are handled differently. Thanks!

Reply:
ah, this is a good question. I changed the condition and unifed them. Hope it is clear now.